### PR TITLE
Fake login for development purposes

### PR DESCRIPTION
**Pull request by hisahi**
### Short description
This adds the ability to do fake logins on a Shibboleth configuration. This is intended for development use and should **under no circumstances** be enabled on a production setup. The feature is controlled by the environment variable `USE_FAKE_LOGIN=1` on the backend and `REACT_APP_USE_FAKE_LOGIN=1` on the frontend; both need to be enabled for the fake login to function. The existing user data used for the fake login is imported from the Kurki simulation test data.

I will add documentation about this feature to the wiki.

### DoD
I have:
- [x] added actual code.
- [x] produced clean code that passes ESLint.
- [x] code that actually does what it should.
- [ ] documented the code or added documentation to the wiki.
- [ ] added or modified test(s).
- [ ] test(s) that actually pass.

### CODE REVIEW

**Comment by gotonode** [comment]
Is the data imported automatically or does that have to be done by running the `npm`-command?

**Comment by gotonode** [comment]
Does this need tests or can this be merged after manual review?

**Comment by hisahi** [comment]
Data is defined under utils/fakeLoginUsers.js and should match the Kurki_simulation data imported to the backend. No tests are necessary other than possibly the fact that the functionality should not at all be changed if the environment variables are set, but this is difficult to automate.

File: backend/server/controllers/login.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ module.exports = {
         }
       }).spread((user, created) => {
         const token = jwt.sign({ username: user.username, id: user.id }, process.env.SECRET)
+        logger.info('stuff', {user, token, created})
```
Comment (mjaakko) [inline backend/server/controllers/login.js]: Maybe change this to more descriptive log message?

Comment (hisahi) [reply backend/server/controllers/login.js]: This shouldn't exist.

**Comment by gotonode** [comment]
This seems to also fix #582 as it informs the user that he/she is using the fake auth system.

**Comment by gotonode** [comment]
I can't remember what WIP means in Git/GitHub (or how to do it).

So I'll just add a note here. Remember to remove `app.use(fakeshibbo)` as it's no longer used (I like using env variables a lot more).

**Pull request merged**